### PR TITLE
ErrorHandling: Hide sensitive data (Mantis 32888) and cleanup code

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Mantis Issue: https://mantis.ilias.de/view.php?id=32888

This PR ...

- ... fixes Mantis issue 32888
- ... adds the correct ILIAS license header (where missing)
- ... applied PSR-2 + X (where missing)
- ... includes a little bit of boyscouting (strict comparisons, obsolete PHPDocs, ...)

### CODE REVIEW

File: Services/Exceptions/classes/class.ilDelegatingHandler.php [hunk 1]
```diff
@@ -51,12 +81,16 @@ public function handle() : ?int
...
         }
-        // we must rest the superglobals back to normal arrays since the error handler needs them. they were replaced by
-        // SuperGlobalDropInReplacement
-        $_GET = (array) $_GET;
-        $_POST = (array) $_POST;
-        $_COOKIE = (array) $_COOKIE;
-        $_REQUEST = (array) $_REQUEST;
+        
+        /* We must cast the superglobals back to normal arrays since the error handler needs them. They were replaced by
+           SuperGlobalDropInReplacement . The keys contain NULL bytes, so accessing values directly by key is not
```
Comment (mjansenDatabay) [inline Services/Exceptions/classes/class.ilDelegatingHandler.php:86]: @chfsx Maybe this is interesting for you (NULL bytes)

File: Services/Exceptions/classes/class.ilDelegatingHandler.php [hunk 2]
```diff
@@ -39,6 +38,37 @@ public function __construct(ilErrorHandling $error_handling)
         $this->error_handling = $error_handling;
     }
 
+    private function hideSensitiveData(array $key_value_pairs) : array
```
Comment (mjansenDatabay) [inline Services/Exceptions/classes/class.ilDelegatingHandler.php:41]: This works for all handlers (plain text, pretty page, ...).

**Review by smeyer-ilias — APPROVED** [review]
Thanks @mjansenDatabay 
LGTM

**Comment by mjansenDatabay** [comment]
@smeyer-ilias Do we really need @pascalseeland here? If you don't want to be assigned I could take over this PR and just wait for some feedback of @chfsx / @klees , who both contributed to the changed files.

**Comment by smeyer-ilias** [comment]
@mjansenDatabay  I removed the assignment, I didn't realize he is on vacation.

**Review by chfsx — APPROVED** [review]

**Comment by mjansenDatabay** [comment]
The TB decided that this PR can be merged by @mjansenDatabay

**Pull request merged**